mycore-base/src/main/java/org/mycore/services/http/MCRQueryParameter.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * This file is part of ***  M y C o R e  ***
+ * See http://www.mycore.de/ for details.
+ *
+ * MyCoRe is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MyCoRe is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MyCoRe.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.mycore.services.http;
+
+import java.net.URLDecoder;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * A record representing a query parameter.
+ *
+ * @see <a href="https://www.w3.org/TR/2014/REC-html5-20141028/forms.html#url-encoded-form-data">W3C recomandation</a>
+ * @param name query parameter name, null is allowed but transformed to empty String
+ * @param value query parameter name, null is allowed but transformed to empty String
+ */
+public record MCRQueryParameter(String name, String value) {
```

Comment:
Is this something we should implement ourselves?
I would prefer the use of an external API for that. For httpclient5 there is a hint to use UriBuilder.
https://stackoverflow.com/a/71477076

Reply:
I would prefer not to mix dependencies. Here this class is used for JDK internal HttpClient. It's small and well tested. Coupling JDK internal HttpClient code with an external HttpClient library (here Apache HttpClient 5.x) would make it to hard to get rid of this dependency later.
IMHO the only advantage of Apache HttpClient at the moment is that's the only implementation thats supports caching. Neither Jetty HttpClient (required for SOLR) nor JDK internalt HttpClient support caching at the moment.

Reply:
Can we rename this to `MCRURLQueryParameter`?
The term *MCRQuery* means something different in the MyCoRe world